tests/drivers/ipm/src/main.c

```diff
@@ -81,24 +81,35 @@ int main(void)
 	int rv, i;
 	const struct device *ipm;
 
+	rv = TC_PASS;
+
 	TC_SUITE_START("test_ipm");
 	ipm = device_get_binding("ipm_dummy0");
-
-	/* Try sending a raw string to the IPM device to show that the
-	 * receiver works
-	 */
-	for (i = 0; i < strlen(thestr); i++) {
-		ipm_send(ipm, 1, thestr[i], NULL, 0);
+	if (ipm == NULL) {
+		printf("ERROR: unable to get device 'ipm_dummy0'\n");
+		rv = TC_FAIL;
+	} else {
+		/* Try sending a raw string to the IPM device to show that the
+		 * receiver works
+		 */
+		int rc = 0;
+
+		for (i = 0; i < strlen(thestr) && rc == 0; i++) {
+			rc = ipm_send(ipm, 1, thestr[i], NULL, 0);
+		}
+		if (rc) {
+			printf("ERROR: ipm_send() error=%u\n", rc);
+			rv = TC_FAIL;
+		} else {
+			/* Now do this through printf() to exercise the sender */
+			/* I will be split to lines of LINE_BUF_SIZE           */
+			printf(LOREM_IPSUM_SHORT "\n");
```

Comment:
```suggestion
			TC_PRINT(LOREM_IPSUM_SHORT "\n");
```

Reply:
it will not work for this test as `TC_PRINT()` resolves to `printk()`